### PR TITLE
feat: add SSH remote log reading support

### LINKED ISSUE #304: SSH remote log reading: ssh://user@host:/path support
## Background

Support reading log files from remote hosts via SSH URL scheme.

**Spec:** `spec/remote-ssh.md`

## Requirements

- [ ] Parse `ssh://[user@]host[:port]:/absolute/path` URL format
- [ ] Establish SSH connection using system SSH config (~/.ssh/config)
- [ ] Key-based auth only (no interactive password prompt)
- [ ] Stream remote file via `cat <path>` into LogStore pipeline
- [ ] Follow mode: use `tail -f` instead of `cat`, keep connection open
- [ ] Connection failure: friendly error message with details
- [ ] Connection drop in follow mode: show `[SSH DISCONNECTED]` in status bar, auto-retry with backoff
- [ ] Source field shows full `ssh://user@host:/path`
- [ ] Mixed local + remote sources in same session
- [ ] Glob patterns in `default_paths` do NOT apply to SSH URLs
- [ ] Config: `ssh.connect_timeout` (default 10s), `ssh.keepalive_interval` (default 30s)

## Acceptance Criteria

- [ ] `scouty-tui ssh://host:/var/log/syslog` loads remote log
- [ ] Follow mode works with remote sources
- [ ] System SSH config respected
- [ ] Mixed local + remote works
- [ ] All existing tests pass

### PR DESCRIPTION
**Pull request by r12f**
## Summary

Add support for reading log files from remote hosts via SSH URL scheme.

### Usage
```bash
scouty-tui ssh://user@host:/var/log/syslog
scouty-tui ssh://host:2222:/var/log/app.log
scouty-tui local.log ssh://prod:/var/log/app.log   # mixed local + remote
```

### Changes

**`loader/ssh.rs`** (new):
- `SshUrl` parser for `ssh://[user@]host[:port]:/absolute/path` format
- `SshLoader` implements `LogLoader` trait — runs `ssh cat <path>` to read remote files
- Key-based auth only (`BatchMode=yes`), no interactive password prompts
- Friendly error messages for common failures (connection refused, permission denied, file not found, hostname resolution)

**`config/mod.rs`**:
- New `SshConfig` section: `connect_timeout` (default 10s), `keepalive_interval` (default 30s)
- SSH URLs in `default_paths` are passed through without glob expansion

**`app.rs`**:
- `load_files_with_ssh()` detects SSH URLs and uses `SshLoader` instead of `FileLoader`
- Mixed local + remote sources in same session

**`main.rs`**:
- Passes `cfg.ssh.connect_timeout` to loader
- Updated help text to mention SSH URLs

### Not Yet Implemented (follow-up)
- Follow mode with `tail -f` over SSH (requires async/streaming architecture)
- Auto-reconnect on SSH disconnection
- Status bar `[SSH DISCONNECTED]` indicator

### Test Plan
- 14 new unit tests for URL parsing, roundtrip, edge cases, shell escaping
- All 524 tests pass

Closes #304

### CODE REVIEW

File: crates/scouty/src/loader/ssh.rs [hunk 1]
```diff
@@ -0,0 +1,242 @@
+//! SSH-based remote log loader — loads log lines from a remote host via SSH.
+
+#[cfg(test)]
+#[path = "ssh_tests.rs"]
+mod ssh_tests;
+
+use crate::traits::{LoaderInfo, LoaderType, LogLoader, Result, ScoutyError};
+use std::process::{Command, Stdio};
```
[the remark this answers is not in the document]
Comment (r12f) [reply crates/scouty/src/loader/ssh.rs]: Already fixed in prior commit — `BufRead` and `BufReader` imports were removed.
